demo/Menu.tsx

```diff
@@ -0,0 +1,72 @@
+import { toggleMark } from "prosemirror-commands";
+import type { MarkType } from "prosemirror-model";
+import type { EditorState } from "prosemirror-state";
+import React, { ReactNode } from "react";
+
+import { useEditorEventCallback, useEditorState } from "../src/index.js";
+
+// lifted from:
+// https://github.com/ProseMirror/prosemirror-example-setup/blob/master/src/menu.ts#L58
+function isMarkActive(mark: MarkType, state: EditorState): boolean {
+  if (!state.selection) return false;
+  const { from, $from, to, empty } = state.selection;
+  return empty
+    ? !!mark.isInSet(state.storedMarks || $from.marks())
+    : state.doc.rangeHasMark(from, to, mark);
+}
+
+export function Button(props: {
+  children?: ReactNode;
+  isActive: boolean;
+  className?: string;
+  onClick: () => void;
+}) {
+  function handleMouseDown(e: React.MouseEvent) {
+    e.preventDefault();
+    props.onClick();
+  }
+  return (
+    <button
+      className={`button ${props.className} ${props.isActive ? "active" : ""}`}
+      onMouseDown={handleMouseDown}
+    >
+      {props.children}
+    </button>
+  );
+}
+
+export default function Menu() {
+  const state = useEditorState();
```

Comment:
just realizing that this is nullable also, i'm guessing for similar reason as [below](https://github.com/nytimes/react-prosemirror/pull/81/files#r1466748866)? not sure why it passed the typechecker, will follow up...

Reply:
https://github.com/nytimes/react-prosemirror/pull/84